colossalai/auto_parallel/solver/strategies_constructor.py

```diff
@@ -349,35 +359,72 @@ def build_strategies_and_cost(self):
+                            output_sharding_spec = ShardingSpec(self.device_mesh,
+                                                                entire_shape_output,
+                                                                dim_partition_dict=dim_partition_dict_for_output)
+                            # TODO: use meta_info_prop to profile origin memory cost and compute cost, then divide them depending on sharding spec.
+                            compute_cost = 0
+                            memory_cost = 0
+                            resharding_costs = generate_resharding_costs(strategies_vector.predecessor_nodes,
+                                                                         [input_sharding_spec])
+                            # to prevent the resharding happening, set their resharding cost to inf.
+                            resharding_costs[input_tensor_node] = [
+                                cost if cost == 0 else math.inf for cost in resharding_costs[input_tensor_node]
+                            ]
+                            sharding_strategy = ShardingStrategy(
+                                name,
+                                output_sharding_spec,
+                                compute_cost=compute_cost,
+                                memory_cost=memory_cost,
+                                resharding_costs=resharding_costs,
+                                input_shardings=[input_tensor_node.output_sharding_spec])
+                            strategies_vector.append(sharding_strategy)
+
+                # torch.arange function
+                elif target == torch.arange:
+                    name = f'FULLY REPLICATED ARANGE'
+                    entire_shape_output = node._meta_data.shape
+                    dim_partition_dict_for_output = {}
+                    output_sharding_spec = ShardingSpec(self.device_mesh,
+                                                        entire_shape_output,
+                                                        dim_partition_dict=dim_partition_dict_for_output)
+                    memory_cost = node._meta_data.numel()
+                    sharding_strategy = ShardingStrategy(name,
+                                                         output_sharding_spec,
+                                                         compute_cost=0,
+                                                         memory_cost=memory_cost)
+                    strategies_vector.append(sharding_strategy)
+
+                # op list to be processed to support gpt2
+                elif target in (builtins.getattr, operator.le, torch.addmm, operator.pow, torch.where, torch.softmax,
+                                torch.nn.functional.softmax, torch.pow, torch.tanh):
+                    pass
                 # other function
                 else:
                     raise RuntimeError(f'{target} function is NOT supported now.')
```

Comment:
This should be offloaded to node handler to produce the strategy. We can do this later.

Reply:
Yes, if we assign an op handler for all kind of op, this should be offloaded. Otherwise, some strategies of simple operations will be generated in StrategiesConstructor.